src/view/actions.ts

```diff
@@ -44,8 +64,29 @@ export async function pushIssue(
 	settings: GitHobsSettings
 ) {
 	if (issueId) {
+		const filePropertiesString = PropertiesHelper.extractFilePropertiesString(file.data);
+		const { properties } = PropertiesHelper.readProperties(filePropertiesString);
+
+		let unsafeFileName = undefined;
+		if (properties) {
+			for (const p of properties) {
+				const propDividerIndex = p.indexOf(': ');
+				const propKey = p.slice(0, propDividerIndex);
+				const propValue = p.slice(propDividerIndex + 2);
+				if (
+					propKey !== PropertiesHelper.GITHUB_ISSUE_PROPERTY_ORIGINAL_TITLE ||
+					!propValue.length
+				)
+					continue;
+				const originalIssueName = propValue.slice(1, propValue.length - 1);
+				if (file.file?.basename === formatUnsafeFileName(originalIssueName)) {
+					unsafeFileName = originalIssueName;
+				}
+			}
+		}
+
```

Comment:
I suggest creating a custom function for this property. Add it in the file `helper/properties.ts`:
```typescript
export function readIssueOriginalTitle(data: string) {
	const { properties } = readProperties(data);
	if (!properties) return;

	const githubIssueOriginalTitleProperty = properties.find((p) =>
		p.startsWith(GITHUB_ISSUE_PROPERTY_ORIGINAL_TITLE)
	);
	if (!githubIssueOriginalTitleProperty) return;

	// Split by first occurrence of :
	const [, originalTitleRaw] = githubIssueOriginalTitleProperty.split(/:(.*)/);
	// Cleanup - Remove all ' and whitespace
	const originalTitle = originalTitleRaw.replace(/'/g, '').trim();

	return originalTitle;
}
```
And use it like this:
```suggestion
const originalIssueName = PropertiesHelper.readIssueOriginalTitle(file.data);

		if (originalIssueName && file.file?.basename === formatUnsafeFileName(originalIssueName)) {
			unsafeFileName = originalIssueName;
		}
```